common.go

```diff
@@ -34,6 +34,10 @@ func getCopyOptions(store storage.Store, reportWriter io.Writer, sourceSystemCon
 	if destinationSystemContext != nil {
 		*destinationCtx = *destinationSystemContext
 	}
+	if destinationCtx != nil {
+		destinationCtx.BigFilesTemporaryDir = sourceCtx.BigFilesTemporaryDir
```

Comment:
This does not conceptually fit. Why is this the business of `getCopyOptions` at all? How would the caller of `getCopyOptions` expect this to happen?

In fact both calls in `commit.go` pass a destination value but not source, so this is clearly not the right place in practice as well.

Reply:
For a start, note that `SystemContext` is not only used for copies.

Then, ideally, trace the full flow of `SystemContext` values and how various fields get set, and try to find a place that fits. Given the uses of `SystemContext` in cmd/buildah, vs. the various code paths that currently don’t use the `SystemContextFromOptions` value at all (just like some of the sources/destinations for `getCopyOptions`), that might involve a larger-scale audit of what the right thing to do is.

Reply:
Your right, I think I have fixed it correctly now.

Reply:
What about all the other code paths?

Reply:
Other code paths are not using dest dir.  Does the big file path get used on pushing?  Or only pulling?

buildah from and I believe buildah bud as well as buildah pull are all using the same pull logic.

Reply:
I just confirmed the buildah bud does use the new code.  So all pull paths are using it.

Reply:
- Commit is a copy from a Buildah-specific source to c/storage, that probably involves the temporary directory
- Other uses of the temporary directory may be added in the future. What’s the user expectation and the right way to implement it? My first impression is that this is (advertised as) a Buildah-general treatment of $TMPDIR that affect absolutely everything.

Reply:
After reviewing the code we are handling the .BigFilesTemporaryDir in those cases, so this was only broken when pulling.

Reply:
the changes to common.go don't seem to be in the pr now, was that intended?

Reply:
Never mind, needed to dig deeper.